### DISCUSSION TITLE
💋 【共学营打卡】【第3次 Milestone】【Move 0x06】Library dApp 实例

### DISCUSSION THREAD

**Discussion by leeduckgo** [NonceGeek/Move-Camp#141] [Projects-Others]
> https://github.com/NonceGeek/Web3-dApp-Camp/tree/main/move-dapp/starcoin/my-library

* 任务 1：成功运行 Library dApp 并截图
* 任务2：查看 library 这个函数在前端显示上可以优化，有哪些优化点？请说明并实践。

**Comment by zhaoge5859** [comment]
任务1打卡
<img width="1128" alt="截屏2022-11-02 12 53 02" src="https://user-images.githubusercontent.com/116735858/199401524-bc79b297-1bda-49b2-8651-2782b405e01b.png">
<img width="529" alt="截屏2022-11-02 12 53 27" src="https://user-images.githubusercontent.com/116735858/199401530-f844b45e-00dc-40a7-9346-4849c388c1cf.png">

**Comment by v1xingyue** [comment]
![image](https://user-images.githubusercontent.com/974169/200279802-50a48c1d-cec6-4f93-8195-adb5bb458385.png)

Counter Library 获取资源均存在bug , 已提交 PR .

**Comment by Demian101** [comment]
<img width="793" alt="image" src="https://user-images.githubusercontent.com/33189338/200460038-312e5785-d630-46bd-a1f8-710c3796891e.png">

增加了一个 Destroy Library 的功能

**Comment by MagicGordon** [comment]
<img width="849" alt="7063" src="https://user-images.githubusercontent.com/19465870/200462885-231c3585-c619-45f5-be0f-8f86172375a9.png">
Pr: #202

**Comment by xiegudong45** [comment]
![image](https://user-images.githubusercontent.com/30063330/201129293-61ffbc7d-a89f-465a-822a-6ba8de6efd3e.png)
![image](https://user-images.githubusercontent.com/30063330/201129522-3631d2a0-49f5-4239-8d3e-00c23a162946.png)

**Comment by livelybug** [comment]
任务1：
![image](https://user-images.githubusercontent.com/13627060/201337687-c22d577d-fe76-4b43-97c8-a2108955d4db.png)

任务2：
增加update_by_index的功能
![image](https://user-images.githubusercontent.com/13627060/201451947-e4d55c0d-e327-4cca-90e6-66eea62ced62.png)

![image](https://user-images.githubusercontent.com/13627060/201451926-2a1ad578-65c6-4ef2-9084-e72bfe102bb9.png)

**Comment by hi-chenerge** [comment]
![1674823317992](https://user-images.githubusercontent.com/82131587/215089806-44903cec-cea3-4e6f-ab0f-175206d22e38.jpg)
![1674823286018](https://user-images.githubusercontent.com/82131587/215089838-01f8925e-2ada-42f5-9931-cca86489bc6d.jpg)
把一些没用的页面元素都删掉了
![1674823395704](https://user-images.githubusercontent.com/82131587/215089970-f9695629-15c3-4f36-8685-a936e94b7406.jpg)

**Comment by baidang201** [comment]
![image](https://user-images.githubusercontent.com/3405789/218303788-276d2b0e-8d01-4c78-a794-b1d4c749546d.png)
![image](https://user-images.githubusercontent.com/3405789/218304347-216175c2-d8b9-4e86-bc9f-821988f0bf22.png)

1 hextoutf8
```
const convert = (from, to) => str => Buffer.from(str, from).toString(to);
const utf8ToHex = convert('utf8', 'hex');
const hexToUtf8 = convert('hex', 'utf8');



const getLibrary = async () => {
    let res = await getResource(LIBRARY_ADDRESS, LIBRARY_RESOURCE_ID)
    console.log(res);
    if (res) {
      for(var i =0; i< res.books.length; i++) {
        let item = res.books[i];
        item.name = hexToUtf8(item.name.substring(2));
        item.link = hexToUtf8(item.link.substring(2));
        res.books[i] = item;
      }
    }

    alert(JSON.stringify(res))
  }
```
